libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/useParentFilters.ts

```diff
@@ -0,0 +1,86 @@
+// (C) 2021 GoodData Corporation
+import { useMemo } from "react";
+import invariant from "ts-invariant";
+import { areObjRefsEqual, filterObjRef, IAttributeFilter, ObjRef } from "@gooddata/sdk-model";
+import { IDashboardAttributeFilter } from "@gooddata/sdk-backend-spi";
+
+import { dashboardAttributeFilterToAttributeFilter } from "../../../_staging/dashboard/dashboardFilterConverter";
+import { selectFilterContextAttributeFilters, useDashboardSelector } from "../../../model";
+import { IAttributeFilterButtonProps } from "@gooddata/sdk-ui-filters";
+
+type UseParentFiltersResult = Pick<
+    IAttributeFilterButtonProps,
+    "parentFilters" | "parentFilterOverAttribute"
+>;
+
+/**
+ * Returns parent filter-related data to use as AttributeFilterButton props.
+ * @param filter - filter to get the parent filter-related data
+ */
+export const useParentFilters = (filter: IDashboardAttributeFilter): UseParentFiltersResult => {
+    const allAttributeFilters = useDashboardSelector(selectFilterContextAttributeFilters);
+
+    // for all parent filters: a map localId -> the filter object
+    // this is then used to derive parentFilterOver attributes
+    const parentFiltersMap = useMemo(() => {
+        if (!filter.attributeFilter.filterElementsBy) {
+            return {};
+        }
+        return filter.attributeFilter.filterElementsBy.reduce(
+            (acc: Record<string, IDashboardAttributeFilter>, curr) => {
+                const matchingFilter = allAttributeFilters.find(
+                    (filter) => filter.attributeFilter.localIdentifier === curr.filterLocalIdentifier,
+                );
+
+                invariant(matchingFilter); // if this blows up, the state is inconsistent
+
+                acc[curr.filterLocalIdentifier] = matchingFilter;
+                return acc;
+            },
+            {},
+        );
+    }, [allAttributeFilters, filter.attributeFilter.filterElementsBy]);
+
+    const parentFilters = useMemo(() => {
+        const values = Object.values(parentFiltersMap);
+        return values.length > 0 ? values.map(dashboardAttributeFilterToAttributeFilter) : undefined;
+    }, [parentFiltersMap]);
+
+    const parentOverLookup = useMemo(() => {
+        // no parents -> no need for the lookup function
+        if (!filter.attributeFilter.filterElementsBy) {
+            return undefined;
+        }
```

Comment:
how about instead using different parentOverLookup function based on this condition?

Reply:
I wanted to return undefined in this case similar to `parentFilters` to make it seem no parent filters were even specified for the AttributeFilterButton (i.e. not to pass the `parentFilterOverAttribute` while not passing the `parentFilters`).